homeassistant/components/ovo_energy/__init__.py

```diff
@@ -33,23 +34,37 @@ async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry) -> bool
     client = OVOEnergy()
 
     try:
-        await client.authenticate(entry.data[CONF_USERNAME], entry.data[CONF_PASSWORD])
+        authenticated = await client.authenticate(
+            entry.data[CONF_USERNAME], entry.data[CONF_PASSWORD]
+        )
+        if not authenticated:
+            hass.async_create_task(
+                hass.config_entries.flow.async_init(
+                    DOMAIN, context={"source": "reauth"}, data=entry.data
+                )
+            )
+            return False
```

Comment:
These likes do not throw the exception and should be pulled out of the try-block.

(another occurrence below).